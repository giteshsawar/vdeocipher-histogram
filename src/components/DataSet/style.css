
.chart {
    /* width: 50%; */
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
    height: 400px;
    margin: 20px;
    padding: 20px;
    border: 1px solid rgba(0, 0, 0, 0.25);
}
.chart .chart_comp {
    width: 100%;
    height: 100;
}
.chart p {
    font-size: 32px;
    color: rgba(0, 0, 0, 0.75);
    line-height: 400px;
    margin: 0;
    cursor: pointer;
}
.chart .chart_comp .field_inputs {
    display: flex;
    justify-content: center;
}
.chart_comp .field_inputs .input_div {
    margin: 0 10px;
}