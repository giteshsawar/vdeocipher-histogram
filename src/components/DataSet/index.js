import React, { useState } from "react";
import ChartComp from "../Chart";
import { getData, calculateTotalCount } from "../../utils/data";

import "./style.css";

function DataSet(props) {
  const { dataset, diff, max } = props;
  const [rawData, setRawData] = useState([]);
  const [chartData, setChartData] = useState([]);
  const [loading, setLoadingStatus] = useState(false);
  const [difference, setDifference] = useState(diff);
  const [maximum, setMaximum] = useState(max);

  const getChartData = async () => {
    setLoadingStatus(true);
    const data = await getData(dataset, difference, maximum);
    if (!data) return window.alert("Not able to fetch data");

    setRawData(data);
    const mappedData = calculateTotalCount(data, {
      dataset,
      diff,
      max,
    });

    setChartData(mappedData);
    setLoadingStatus(false);
  };

  const getInputValue = e => {
    const { name, value } = e.target;
    const num = Number(value);
    console.log('set new value', num, typeof num);
    if (name === 'max') return setMaximum(num);

    if (num <= maximum) return setDifference(num);

    alert('Difference must be smaller than maximum');
  };

  const updateChart = () => {
    const mappedData = calculateTotalCount(rawData, { dataset, diff: difference, max: maximum });
    setChartData(mappedData);
  };

  return (
    <div className="chart">
      {chartData.length ? (
        <div className="chart_comp" id="chart_container">
          <div className="field_inputs">
            <div className="input_div">
              Difference: <input type="number" name="diff" value={difference} onChange={getInputValue} />
            </div>
            <div className="input_div">
              Max: <input type="number" name="max" value={maximum} onChange={getInputValue} />
            </div>
            <div className="input_div">
              <button className="btn" onClick={updateChart}>Update</button>
            </div>
          </div>
          <ChartComp data={chartData} />
        </div>
      ) : (
        <div>
          {loading ? (
            <p id="laoding_msg">Loading data...</p>
          ) : (
            <p id="fetch_data" onClick={getChartData}>Click here to get {dataset} data</p>
          )}
        </div>
      )}
    </div>
  );
}

export default DataSet;
