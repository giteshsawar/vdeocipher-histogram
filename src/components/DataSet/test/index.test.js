// Link.react.test.js
import React from "react";
import DataSet from "../index";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Dataset testing", () => {
  let component;
  beforeEach(() => {
    component = shallow(<DataSet />);
  });

  it("snapshot match", () => {
    expect(component).toMatchSnapshot();
  });

  it("find button to fetch data", () => {
    expect(component.find("#fetch_data"));
  });

  it("loading data on click", () => {
    window.alert = () => {};
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    });
    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

    component.find("#fetch_data").simulate("click");
    expect(global.fetch).toHaveBeenCalledTimes(1);

    expect(component.find("#laoding_msg"));
  });
});
