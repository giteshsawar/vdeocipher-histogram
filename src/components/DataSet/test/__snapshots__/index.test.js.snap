// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dataset testing snapshot match 1`] = `ShallowWrapper {}`;

exports[`Link changes the class when hovered 1`] = `ShallowWrapper {}`;

exports[`snapshot match 1`] = `ShallowWrapper {}`;
